example-service/src/client.rs

```diff
@@ -38,20 +39,91 @@ async fn main() -> io::Result<()> {
         .parse::<SocketAddr>()
         .unwrap_or_else(|e| panic!(r#"--server_addr value "{}" invalid: {}"#, server_addr, e));
 
-    let name = flags.value_of("name").unwrap().into();
+    let num_connections: usize = flags
+        .value_of("num_connections")
+        .unwrap()
+        .parse()
+        .expect("num_connections needs an argument");
+
+    let (senders, receivers): (
+        Vec<tokio::sync::mpsc::Sender<Instant>>,
+        Vec<tokio::sync::mpsc::Receiver<Instant>>,
+    ) = (0..num_connections)
+        .map(|_| tokio::sync::mpsc::channel(ROUNDS + 1))
+        .unzip();
+
+    for mut sender in senders.into_iter() {
+        tokio::spawn(async move {
+            sender
+                .send(Instant::now())
+                .await
+                .expect("failed to send instant");
```

Comment:
The expect message is only printed in case of failure, so I think the "failed to" prefix is redundant. Also, instead of "instant", perhaps include the purpose of the thing being sent (start time)?

```suggestion
                .expect("send start connecting time");
```